### PR TITLE
Read with underscore

### PR DESCRIPTION
**Pull request by alexrockhill**
Thanks for contributing. If this is your first time,
make sure to read [contributing.md](https://github.com/mne-tools/mne-bids/blob/master/CONTRIBUTING.md)

PR Description
--------------

Don't include BIDS directory name in parsing because it can cause issues.

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"
- [ ] Commit history does not contain any merge commits

### CODE REVIEW

File: mne_bids/tests/test_read.py [hunk 1]
```diff
@@ -24,6 +24,9 @@
     subject=subject_id, session=session_id, run=run, acquisition=acq,
     task=task)
 
+def test_read_raw_bids():
+    output_path, bids_basename, raw, raw_fname = write_test_bids()
+    read_raw_bids(bids_basename, output_path)
```
Comment (sappelhoff) [inline mne_bids/tests/test_read.py]: with this test, you are not testing the bugfix: you are passing `bids_basename`, which is the following:

```Python
bids_basename = make_bids_basename(
    subject=subject_id, session=session_id, run=run, acquisition=acq,
    task=task)
```
This yields a NAME (not a path), as you can see in [this example](https://mne-tools.github.io/mne-bids/auto_examples/create_bids_folder.html#creating-file-names-for-bids). Note that I am aware that the distinction between name and path is confusing ... and it's great that you are submitting a fix to this.

See my comment on the issue: https://github.com/mne-tools/mne-bids/issues/230#issuecomment-512188461

To actually test your bugfix, you should pass a **path**:

this fails on master: `bids_fname = '/home/stefanappelhoff/Downloads/eeg_matchingpennies/sub-05/eeg/sub-05_task-matchingpennies_eeg.vhdr'` ... because master expects `bids_fname='sub-05_task-matchingpennies_eeg.vhdr'`

but passing the whole path on your branch will work due to your `op.basename` fix

Long story short: just make a very simple and tiny test by passing a `bids_fname` that is a path ... instead of reusing the `bids_basename` that is defined at the module level

**Comment by sappelhoff** [comment]
Just a note: It'd be great if you could edit your original post to include:

`closes #230`, because then upon merge of this PR, your issue will be automatically closed.

Also, you can delete the following:
> Thanks for contributing. If this is your first time, make sure to read contributing.md

**Comment by jasmainak** [comment]
@alexrockhill seems like there is a cleaner fix in https://github.com/mne-tools/mne-bids/pull/233/files. Do you want to give it a try and see if it works for you?
